src/Symfony/Component/HttpKernel/DataCollector/RequestDataCollector.php

```diff
@@ -276,18 +250,37 @@ public function getRouteParams()
     }
 
     /**
-     * Gets the controller.
+     * Gets the parsed controller.
      *
-     * @return string The controller as a string
+     * @return array|string The controller as a string or array of data
+     *                      with keys 'class', 'method', 'file' and 'line'
      */
     public function getController()
     {
         return $this->data['controller'];
     }
 
+    /**
+     * Gets the parsed forward controller.
+     *
+     * @return array|bool The parsed forward controller as an array of data
+     *                    with keys 'token' the forward profile token, and
```

Comment:
It's nice to have good comments but I think you are revealing too many implementation details in this one.